srt-protocol/src/protocol/receiver/buffer.rs

```diff
@@ -262,9 +265,13 @@ impl ReceiveBuffer {
         let timestamp = match self.front_ts() {
             Some(timestamp) => timestamp,
             None => {
-                return match self.drop_too_late_packets(now) {
-                    Some(error) => Err(error),
-                    None => Ok(None),
+                if self.too_late_packet_drop {
+                    return match self.drop_too_late_packets(now) {
```

Comment:
DRY this code: instead of repeating the `self.too_late_packet_drop` check, move it inside the `drop_too_late_packets` function